OpenVino 2019 R1 - OpenCV4 Support

##### System information (version)

- OpenCV => 4.0.1
- Operating System / Platform => Mac


##### Detailed description

Openvino 2019 R1 is available for mac now. 
I am trying to build opencv from source with Inference Engine. But the build fails while compiling dnn module. When will the latest version be supported ?

Thanks in advance.

Thanks @dkurt and @alalek . Now I can build from master branch. But I am facing this issue though. 

/tmp/open/opencv-master/modules/dnn/src/op_inf_engine.cpp:811: error: (-215:Assertion failed) Failed to initialize Inference Engine backend: Unsupported primitive of type: PriorBoxClustered name: PriorBox_5 in function 'initPlugin'

May be this might have something to do with this error. I am using mac.
https://github.com/opencv/opencv/blob/master/modules/dnn/src/op_inf_engine.cpp#L785

It works for simple models which doesn't use any extra layer libraries.

--- a/modules/dnn/src/op_inf_engine.cpp
+++ b/modules/dnn/src/op_inf_engine.cpp
@@ -784,6 +784,8 @@ void InfEngineBackendNet::initPlugin(InferenceEngine::ICNNNetwork& net)
                         continue;
     #ifdef _WIN32
                     std::string libName = "cpu_extension" + suffixes[i] + ".dll";
+    #elif defined(__APPLE__)
+                    std::string libName = "libcpu_extension" + suffixes[i] + ".dylib";
     #else
                     std::string libName = "libcpu_extension" + suffixes[i] + ".so";
     #endif  // _WIN32

--- a/modules/dnn/test/test_ie_models.cpp
+++ b/modules/dnn/test/test_ie_models.cpp
@@ -172,6 +172,8 @@ void runIE(Target target, const std::string& xmlPath, const std::string& binPath
                     continue;
 #ifdef _WIN32
                 std::string libName = "cpu_extension" + suffixes[i] + ".dll";
+#elif defined(__APPLE__)
+                std::string libName = "libcpu_extension" + suffixes[i] + ".dylib";
 #else
                 std::string libName = "libcpu_extension" + suffixes[i] + ".so";
 #endif  // _WIN32

